stablehlo/reference/Ops.cpp

```diff
@@ -78,6 +78,12 @@ Tensor evalBroadcastInDimOp(const Tensor &operand,
   return result;
 }
 
+SmallVector<Tensor> evalCaseOp(const Tensor &index,
+                               MutableArrayRef<Region> branches, Scope &scope) {
+  return eval(branches[index.get({}).getIntegerValue().getSExtValue()], {},
```

Comment:
maybe we should check for index out of bound (>= branches.size()). If yes, then the `branches[branches.size()-1]` is chosen.